tutorial/cache_pyh_files.md

```diff
@@ -0,0 +1,32 @@
+# Picklize Header files
+
+## Header files
+
+A header file in Pyccel is a file containing function/variable declarations, macro definitions , templates and metavariable declarations.\
+the rule is to give header files names that end with `.pyh` .\
+Header files serve two purposes:
+-  Link external libraries in the targeted languages by providing their function definitions.
+-  Accelerate the parsing process by parsing the header file instead of the original file in the case of pyccelizing multiple files.
+
+### Example
+We create the file `header.pyh` that contains an openmp function definiton:
+
+```python
+#$ header metavar module_name = 'omp_lib'
+#$ header metavar import_all  = True
+
+#$ header function omp_get_num_threads() results(int)
+```
+We then create `openmp.py` file:
+
+```python
+from header import omp_get_num_threads
+print('number of threads is :', omp_get_num_threads())
+```
+Pyccel can compile the Python file with the following command: `pyccel openmp.py --language fortran --flags="-fopenmp"`
+, It will then create the executable file `openmp`
+
+## Picklizing header files
+Pyccel uses the Python Module [pickle](https://docs.python.org/3/library/pickle.html) to cache the header files.\
+When compiling a header file Pyccel will generate in the same directory a `.pyccel` file that contains the cached result of the parser,\
+This will accelerate the compiling process of big header files, by compiling them only once and storing the results for future compilation, Pyccel will generate a new `.pyccel` when after the modification the header file  or downloading a new Pyccel version.
```

Comment:
We should clarify the language a bit here. For example, say that Pyccel is "processing" a file instead of "compiling" them. Otherwise, we should say at the beginning what we mean by "compiling".

Reply:
it's the same the difference is in the code generation

Reply:
I agree. For me "compiling" means using a compiler. e.g gcc
I would use either "processing" or "pyccelizing"

Reply:
> it's the same the difference is in the code generation

?

Reply:
I mean Pyccel is a compiler , so what it does is compiling python code and generating a `.so`, even for c programs , the c compiler generates assembly code and then the assembly compiles it to generate machine code

Reply:
I am going to change the explanation myself. Let me know if I write anything wrong.

Reply:
okay